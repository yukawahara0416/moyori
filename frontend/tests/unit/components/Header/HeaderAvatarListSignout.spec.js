import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Component from '@/components/Header/HeaderAvatarListSignout.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper
let store

const signOutHandler = jest.fn()

beforeEach(() => {
  store = new Vuex.Store()

  wrapper = mount(Component, {
    localVue,
    store,
    methods: {
      signOutHandler
    }
  })
})

describe('getters', () => {
  it('headers', () => {})
})

describe('v-on', () => {
  it('signOutHandler', () => {
    wrapper.find('.v-list-item').trigger('click')
    expect(signOutHandler).toHaveBeenCalledTimes(1)
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
