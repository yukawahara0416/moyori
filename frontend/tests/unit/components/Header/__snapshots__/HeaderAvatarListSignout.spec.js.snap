// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div
  class="v-list-item v-list-item--link theme--light"
  tabindex="0"
>
  <div
    class="v-avatar v-list-item__avatar"
    style="height: 40px; min-width: 40px; width: 40px;"
  >
    <i
      aria-hidden="true"
      btn=""
      class="v-icon notranslate mdi mdi-logout theme--light"
    />
  </div>
   
  <div
    class="v-list-item__content"
  >
    <div
      class="v-list-item__title"
    >
      ログアウト
    </div>
  </div>
</div>
`;
